### ISSUE TITLE
When using the same FileReader instance across multiple threads, the threads seem to hang

### ISSUE THREAD

**Issue by menacher** [IPQualityScore/JavaIPQSDBReader#2]
Here is a unit test which shows the scenario.
```
@Test
    public void microBenchMarkWithSameFileHandler() throws IOException, InterruptedException {
        ExecutorService pool = Executors.newFixedThreadPool(25);
        String fileName = "IPQualityScore-IP-Reputation-Database-IPv4.ipqs";
        File f = new File(getClass().getClassLoader().getResource(fileName).getFile());
        String ip = "8.8.0.0";
        FileReader reader = DBReader.Open(f.getAbsolutePath());
        double loop = 1000;
        long start = System.currentTimeMillis();
        CountDownLatch latch = new CountDownLatch((int)loop);
        for (int i = 0; i < loop; i++) {
            pool.submit(() -> {
                try {
                    assertNotNull(reader.Fetch(ip));
                } catch (IOException e) {
                    throw new RuntimeException(e);
                }
                latch.countDown();
                System.out.println(latch.getCount());
            });
        }
        System.out.println("Finished submitting read requests in " + (System.currentTimeMillis() - start) + "ms");
        boolean await = latch.await(60, TimeUnit.SECONDS);
        assertTrue(await);
        long totalTime = System.currentTimeMillis() - start;
        System.out.println("Finished waiting for requests in " + totalTime + "ms");
        System.out.printf("Avg read request time %f%s%n", totalTime/loop, "ms");
        reader.getHandler().close();
    }
```
All the threads seem to get parked here. This issue does not occur if I do `DBReader.Open` inside every thread.
```
"pool-1-thread-1" #20 prio=5 os_prio=31 cpu=5.10ms elapsed=12.29s tid=0x0000000107279800 nid=0x9203 waiting on condition  [0x000000016e7fa000]
   java.lang.Thread.State: WAITING (parking)
	at jdk.internal.misc.Unsafe.park(java.base@11.0.15/Native Method)
	- parking to wait for  <merged>(a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.park(java.base@11.0.15/LockSupport.java:194)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(java.base@11.0.15/AbstractQueuedSynchronizer.java:2081)
	at java.util.concurrent.LinkedBlockingQueue.take(java.base@11.0.15/LinkedBlockingQueue.java:433)
	at java.util.concurrent.ThreadPoolExecutor.getTask(java.base@11.0.15/ThreadPoolExecutor.java:1054)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@11.0.15/ThreadPoolExecutor.java:1114)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@11.0.15/ThreadPoolExecutor.java:628)
	at java.lang.Thread.run(java.base@11.0.15/Thread.java:829)
```

**Comment by IPQualityScore** [comment]
We would encourage you to open a new instance per thread. The overhead is minimal and all of our unit tests involving multithreading presume a single instance per thread. Is there a reason you would prefer to open a single instance?

**Comment by menacher** [comment]
![Screen Shot 2022-08-19 at 3 14 16 PM](https://user-images.githubusercontent.com/1030178/185690924-8cd00322-9b50-437b-b20a-f413b1510566.jpg)
When I profiled it, the DBReader.Open is taking ~70% of the execution time vs fetch which takes ~30%. Hence I reported it.
![Screen Shot 2022-08-19 at 3 15 27 PM](https://user-images.githubusercontent.com/1030178/185691076-c06c41ec-e899-4798-a11c-fda9c73e9805.jpg)

**Comment by IPQualityScore** [comment]
We've made modifications to this to hopefully resolve this issue by using a channel reader.
